sliceop/f64s/f64s.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Common operations on float64 slices
+package f64s
+
+
+// Return a slice with all elements for which f(x) returns true
+func Filter(dst, src []float64, f func(v float64) bool) []float64 {
+	if dst == nil {
+		dst = []float64{}
+	}
+	for _, x := range src {
+		if f(x) {
+			dst = append(dst, x)
+		}
+	}
+	return dst	
+}
+
+// Return a slice of {f(x_i)} when src = {x_i}
+func Map(dst, src []float64, f func(v float64) float64) []float64 {
+	if dst == nil {
+		dst = []float64{}
+	}
+	for _, x := range src {
```

Comment:
add, before the loop:
```go
if len(src) != len(dst) {
    panic("f64s: length mismatch")
}
```

this should also help the compiler to automatically remove a few bound-check tests.

then, I'd rewrite the for-loop like so:
```go
for i, v := range src {
    dst[i] = f(v)
}
```